src/analytic/DDSR2D/DDSR2DReconstruction.cxx

```diff
@@ -0,0 +1,586 @@
+    }
+#endif
+
+}
+
+bool DDSR2DReconstruction::post_processing()
+{
+  return base_type::post_processing();
+}
+
+Succeeded
+DDSR2DReconstruction::
+set_up(shared_ptr <DDSR2DReconstruction::TargetT > const& target_data_sptr)
+{
+  if (base_type::set_up(target_data_sptr) == Succeeded::no)
+    return Succeeded::no;
+
+	if (attenuation_map_filename.length() == 0) 
+     error(boost::format("You need to specify an attenuation map file"));
+
+	if (noise_filter > 1)
+     error(boost::format("Noise filter has to be between 0 and 1 but is %g") % noise_filter);    
+
+	if (noise_filter2 > 1)
+     error(boost::format("Noise filter 2 has to be between 0 and 1 but is %g") % noise_filter2);    
+	
+	// TODO improve this, drop "stir/IO/read_from_file.h" if possible
+	atten_data_ptr= read_from_file<DiscretisedDensity<3,float> >(attenuation_map_filename);
+	
+  if (num_segments_to_combine>=0 && num_segments_to_combine%2==0)
+    error(boost::format("num_segments_to_combine has to be odd (or -1), but is %d") % num_segments_to_combine);
+
+    if (num_segments_to_combine==-1)
+    {
+      const shared_ptr<const ProjDataInfoCylindrical> proj_data_info_cyl_sptr =
+	dynamic_pointer_cast<const ProjDataInfoCylindrical>(proj_data_ptr->get_proj_data_info_sptr());
+
+      if (is_null_ptr(proj_data_info_cyl_sptr))
+        num_segments_to_combine = 1; //cannot SSRB non-cylindrical data yet
+      else
+	{
+	  if (proj_data_info_cyl_sptr->get_min_ring_difference(0) !=
+	      proj_data_info_cyl_sptr->get_max_ring_difference(0)
+	      ||
+	      proj_data_info_cyl_sptr->get_num_segments()==1)
+	    num_segments_to_combine = 1;
+	  else
+	    num_segments_to_combine = 3;
+	}
+    }
+
+
+  return Succeeded::yes;
+}
+
+std::string DDSR2DReconstruction::method_info() const
+{
+  return "DDSR2D";
+}
+
+DDSR2DReconstruction::
+DDSR2DReconstruction(const std::string& parameter_filename)
+{  
+  initialise(parameter_filename);
+  //std::cerr<<parameter_info() << std::endl;
+  info(boost::format("%1%") % parameter_info());
+}
+
+DDSR2DReconstruction::DDSR2DReconstruction()
+{
+  set_defaults();
+}
+
+DDSR2DReconstruction::
+DDSR2DReconstruction(const shared_ptr<ProjData>& proj_data_ptr_v,  
+			const shared_ptr<DiscretisedDensity<3,float> >& atten_data_ptr_v,
+			const double noise_filter_v,  
+			const double noise_filter2_v, 
+		    const int num_segments_to_combine_v
+)
+{
+  set_defaults();
+
+  num_segments_to_combine = num_segments_to_combine_v;
+  proj_data_ptr = proj_data_ptr_v;
+  atten_data_ptr = atten_data_ptr_v; 
+  noise_filter = noise_filter_v; 
+  noise_filter2 = noise_filter2_v; 
+  // have to check here because we're not parsing
+//  if (post_processing_only_DDSR2D_parameters() == true)
+//    error("DDSR2D: Wrong parameter values. Aborting\n");
+} 
+
+Succeeded 
+DDSR2DReconstruction::
+actual_reconstruct(shared_ptr<DiscretisedDensity<3,float> > const & density_ptr)
+{
+
+  // perform SSRB
+ /* if (num_segments_to_combine>1)
```

Comment:
commented out ATM, I suppose because of SPECT. Best to remove the code completely. then we shouldn't have the `num_segments_to_combine` keyword either.